### ISSUE TITLE
weston: panic in KbState::init_with_fd (alacritty)

### ISSUE THREAD

**Issue by dnkl** [Smithay/client-toolkit#107]
When running alacritty master (https://github.com/alacritty/alacritty/commit/14dc170caafc47a46d4c040a17829b52d431bbcd) on **Weston-8.0.0**, I get a panic:
<details><summary>Backtrace</summary>

```sh
RUST_BACKTRACE=full ./alacritty/target/debug/alacritty
```
```sh
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 1, kind: PermissionDenied, message: "Operation not permitted" }', src/libcore/result.rs:1188:5
stack backtrace:
   0:     0x55f9e1395f64 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h7b11b2a23cd910b3
   1:     0x55f9e13b966c - core::fmt::write::hca7efce6c0791439
   2:     0x55f9e1392ac7 - std::io::Write::write_fmt::he56b7f6fb86cf3b5
   3:     0x55f9e139822e - std::panicking::default_hook::{{closure}}::hf51e7f0a7424f5cd
   4:     0x55f9e1397f21 - std::panicking::default_hook::he8ba9c8182908cf7
   5:     0x55f9e139890b - std::panicking::rust_panic_with_hook::hf9a08396e191b616
   6:     0x55f9e13984be - rust_begin_unwind
   7:     0x55f9e13b5a6e - core::panicking::panic_fmt::h1ec1b70c49f8bf0e
   8:     0x55f9e13b5b67 - core::result::unwrap_failed::h9d4b97b9a9041349
   9:     0x55f9e11bb6e3 - core::result::Result<T,E>::unwrap::h461ed369cb24ece9
                               at /build/rust/src/rustc-1.41.0-src/src/libcore/result.rs:956
  10:     0x55f9e114c43e - smithay_client_toolkit::keyboard::KbState::init_with_fd::hb7b864f674677262
                               at /home/daniel/.cargo/registry/src/github.com-1ecc6299db9ec823/smithay-client-toolkit-0.6.6/src/keyboard/mod.rs:302
  11:     0x55f9e0f94222 - <smithay_client_toolkit::keyboard::KbdHandler<Impl,RepeatImpl> as wayland_client::protocol::wl_keyboard::EventHandler>::keymap::hcf78d802990a4f5b
                               at /home/daniel/.cargo/registry/src/github.com-1ecc6299db9ec823/smithay-client-toolkit-0.6.6/src/keyboard/mod.rs:834
  12:     0x55f9e0fa1aef - <wayland_client::protocol::wl_keyboard::WlKeyboard as wayland_client::proxy::HandledBy<T>>::handle::hf6902ab43d9c5639
                               at /home/daniel/src/alacritty/target/debug/build/wayland-client-5043a85dfcd418bb/out/wayland_api.rs:7523
  13:     0x55f9e0f6c1e7 - wayland_client::proxy::NewProxy<I>::implement::{{closure}}::h76b0197b36b56539
                               at /home/daniel/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.23.6/src/proxy.rs:369
  14:     0x55f9e0f9973b - <alloc::boxed::Box<F> as core::ops::function::FnMut<A>>::call_mut::h0a8530b4628d3528
                               at /build/rust/src/rustc-1.41.0-src/src/liballoc/boxed.rs:1029
  15:     0x55f9e0fd19b3 - wayland_client::imp::proxy::proxy_dispatcher::{{closure}}::hc1e623a00100b545
                               at /home/daniel/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.23.6/src/native_lib/proxy.rs:369
```
</details>

**Comment by vberger** [comment]
Interestingly, I cannot reproduce this issue with the SCTK examples on `v0.6.x`, while I can reproduce the alacritty crash. This is pretty surprising as they should be using the same code.

**Comment by vberger** [comment]
Ah, got it. This is actually a [`smithay-clipboard`](https://github.com/Smithay/smithay-clipboard) bug, which binds `wl_seat` without actually checking the maximum version. As a result it binds version 7 which changes the semantics of memmap the clients must use.

**Comment by vberger** [comment]
I released smithay-clipboard version 0.3.7 which fixes this crash. `cargo update` should bring it to you.

**Comment by dnkl** [comment]
That did it, thanks for the quick fix!
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
